### PR TITLE
Document E2E encryption.

### PR DESCRIPTION
**Pull request by htroisi**

### CODE REVIEW

File: content/en/01-about-pixie/01-what-is-pixie.md [hunk 1]
```diff
@@ -7,32 +7,34 @@ redirect_from:
...
 
-## Data Sources
+collects in-memory on the nodes in your cluster
 
-When Pixie is deployed on a Kubernetes cluster, it uses eBPF to automatically collect a rich set of data sources. These include full-body requests traces, resource and network metrics, and application profiles.
+* **Scriptability**: [PxL](/reference/pxl/), Pixie’s flexible Pythonic query language, can be used across Pixie’s UI, CLI, and client APIs. Pixie provides many [community scripts](https://github.com/pixie-labs/pixie/tree/main/src/pxl_scripts) that enable developers  to use Pixie for a broad set of use cases out of the box. Check out the [tutorials](/tutorials/pixie-101) to see how to use Pixie for network monitoring, infrastructure health, service performance, database query profiling, request tracing and continuous application profiling.
 
-A full list of Pixie's supported datasources can be found [here](/about-pixie/data-sources).
+## Architecture
```
Comment (nserrino) [inline content/en/01-about-pixie/01-what-is-pixie.md]: I think these are too wordy now

**Review by nserrino — COMMENTED** [review]
This looks good to me but someone who was part of e2e encryption should take a pass.

Also, there is a lot of stuff in here not related to e2e encryption at all. In the future we should separate that out into a separate PR

File: content/en/03-using-pixie/02-using-cli.md [hunk 2]
```diff
@@ -142,7 +144,23 @@ px run -f <script.pxl>		# Use Pixie's CLI to run the pxl script with the provide
 px live -f <script.pxl>		# Use Pixie's Live CLI to run the pxl script with the provided filename
 ```
 
-## Debugging Pixie using the CLI
+## Advanced
+
+### Disabling Encryption
```
Comment (vihangm) [inline content/en/03-using-pixie/02-using-cli.md]: Just title this `End-to-End Encryption`

File: content/en/03-using-pixie/02-using-cli.md [hunk 3]
```diff
@@ -142,7 +144,23 @@ px run -f <script.pxl>		# Use Pixie's CLI to run the pxl script with the provide
...
+
+By default, Pixie implements end-to-end encryption for telemetry data requested by the CLI. For more information, see the [FAQ](/about-pixie/faq#how-does-pixie-secure-its-data).
```
Comment (vihangm) [inline content/en/03-using-pixie/02-using-cli.md]: Do we call it `telemetry data` elsewhere? I called it `script data` in our API release.
Also drop the `By default` here since the defaults might change and we probably want to publish this independently of when we change the default behavior.

File: content/en/03-using-pixie/02-using-cli.md [hunk 4]
```diff
@@ -142,7 +144,23 @@ px run -f <script.pxl>		# Use Pixie's CLI to run the pxl script with the provide
...
+
+For faster script execution, opt out of E2E encryption for a specific query using the `--disable_e2e_encryption` flag:
```
Comment (vihangm) [inline content/en/03-using-pixie/02-using-cli.md]: Use the `--e2e_encryption` flag to control encryption of table data. Enabling E2E encryption adds some overhead to encrypt the results on the Vizier side and decrypt it on the CLI side.

File: content/en/03-using-pixie/03-api-quick-start.md [hunk 5]
```diff
@@ -283,3 +283,9 @@ func (s *tableMux) AcceptTable(ctx context.Context, metadata types.TableMetadata
 	return &tablePrinter{}, nil
 }
 ```
+
+## Enabling Encryption
```
Comment (vihangm) [inline content/en/03-using-pixie/03-api-quick-start.md]: Change this to `End-to-End Encryption`

File: content/en/03-using-pixie/03-api-quick-start.md [hunk 6]
```diff
@@ -283,3 +283,9 @@ func (s *tableMux) AcceptTable(ctx context.Context, metadata types.TableMetadata
...
+
+Pixie offers end-to-end encryption for telemetry data requested by the API. For more information, see the [FAQ](/about-pixie/faq#how-does-pixie-secure-its-data).
+
+Encryption is enabled with a flag during client creation. For additional info, see the [encryption example](https://github.com/pixie-labs/pixie/blob/main/src/api/go/pxapi/examples/encryption_example/example.go#L77) project.
```
Comment (vihangm) [inline content/en/03-using-pixie/03-api-quick-start.md]: `controlled by a flag`

File: content/en/03-using-pixie/02-using-cli.md (hunk 3 above)
Comment (htroisi) [reply content/en/03-using-pixie/02-using-cli.md]: We currently use telemetry data throughout the docs (no instances of "script data"), so I'll keep it consistent unless you strongly prefer else

File: content/en/01-about-pixie/04-faq.md [hunk 7]
```diff
@@ -43,47 +45,73 @@ order: 4
...
 Retention time depends on the level of traffic in your cluster, but will generally be on the order of hours. Each data table in Pixie (e.g. `http_events`) has its own maximum size. These tables collect data until the maximum size is reached, and then begin dropping the oldest data.
 
-We recommend integrating with third-party observability tools (such as New Relic’s integration) to provide long-term retention.
+We recommend integrating with third-party observability tools (such as [New Relic’s integration](https://newrelic.com/platform/kubernetes-pixie)) to provide long-term retention.
+
+### How does Pixie secure its data?
+
+Pixie stores the telemetry data it collects in-memory on the nodes in your cluster. Data processing and script execution are also performed in the cluster. This is true for both self-hosted Pixie and [Pixie Community Cloud](/about-pixie/faq#does-pixie-offer-a-hosted-cloud-offering).
+
+End-to-end encryption is offered for data in flight between in-cluster storage and presentation in the UI / CLI / API:
```
Comment (htroisi) [inline content/en/01-about-pixie/04-faq.md]: @vihangm - did you see this section? Should I update the Live UI / CLI's enabled statuses to reflect the current state or the intended near-future state?

File: content/en/01-about-pixie/04-faq.md [hunk 8]
```diff
@@ -43,47 +45,73 @@ order: 4
...
+Pixie has a [2GiB memory requirement](/installing-pixie/requirements/#memory) per node. After installing Pixie, it is normal to see a temporary increase in memory usage of the `vizier-pem` pods as they begin to fill their data tables.
+
+### How much data does Pixie store?
 
 Retention time depends on the level of traffic in your cluster, but will generally be on the order of hours. Each data table in Pixie (e.g. `http_events`) has its own maximum size. These tables collect data until the maximum size is reached, and then begin dropping the oldest data.
 
-We recommend integrating with third-party observability tools (such as New Relic’s integration) to provide long-term retention.
+We recommend integrating with third-party observability tools (such as [New Relic’s integration](https://newrelic.com/platform/kubernetes-pixie)) to provide long-term retention.
+
+### How does Pixie secure its data?
```
Comment (htroisi) [inline content/en/01-about-pixie/04-faq.md:92]: @aimichelle - Could you review this section? Specifically, the _Community Cloud for Pixie Security_ blurb at the bottom.

File: content/en/03-using-pixie/02-using-cli.md (hunk 3 above)
Comment (vihangm) [reply content/en/03-using-pixie/02-using-cli.md]: Consistency sounds good

File: content/en/01-about-pixie/04-faq.md [hunk 9]
```diff
@@ -43,47 +45,73 @@ order: 4
...
+
+Pixie stores the telemetry data it collects in-memory on the nodes in your cluster. Data processing and script execution are also performed in the cluster. This is true for both self-hosted Pixie and [Pixie Community Cloud](/about-pixie/faq#does-pixie-offer-a-hosted-cloud-offering).
+
+End-to-end encryption is offered for data in flight between in-cluster storage and presentation in the UI / CLI / API:
+
+- **Live UI** (enabled by default)
```
Comment (vihangm) [inline content/en/01-about-pixie/04-faq.md]: let's drop the defaults from this section so that we don't have to edit it.

File: content/en/03-using-pixie/02-using-cli.md [hunk 10]
```diff
@@ -142,7 +144,25 @@ px run -f <script.pxl>		# Use Pixie's CLI to run the pxl script with the provide
...
+Pixie offers end-to-end encryption for telemetry data requested by the CLI. For more information, see the [FAQ](/about-pixie/faq#how-does-pixie-secure-its-data).
+
+Encryption is controlled by the `--e2e_encryption` flag during script execution. Enabling E2E encryption adds some overhead to encrypt the results on the Vizier side and decrypt it on the CLI side.
+
+To opt out of E2E encryption for a specific query:
+
+```bash
+# Disable encryption when running a script with the CLI.
+px run <script_name> --e2e_encryption=false
+
```
Comment (vihangm) [inline content/en/03-using-pixie/02-using-cli.md]: maybe add one example to enable it?

File: content/en/01-about-pixie/04-faq.md [hunk 11]
```diff
@@ -43,47 +45,73 @@ order: 4
...
+- **API** (disabled by default, [enable during client creation](/using-pixie/api-quick-start/#encryption))
+
+Pixie Cloud (self-hosted or Pixie Community Cloud) hosts the UI and stores limited metadata related to account (user, organization) and Kubernetes control data (cluster name, number of nodes, etc). All communication with Pixie Cloud is TLS encrypted.
+
+Pixie supports two modes for accessing data by the UI. In Passthrough Mode, data flows through Pixie's Cloud via a reverse proxy as encrypted traffic without any persistence. This convenience mode allows developers to access data without being in the same network as the cluster. In Data Isolation Mode, no data flows through Pixie Cloud, instead the browser directly proxies into Pixie's Vizier Module. For more info, see [Configuring Data Transfer Modes](/reference/admin/data-transfer-mode/).
+
+_Community Cloud for Pixie Security_
+
+Pixie Community Cloud is a hosted version of Pixie Cloud. Customer metadata will only be accessed by a limited production team for support purposes. All access is logged in an internal tool.
```
Comment (aimichelle) [inline content/en/01-about-pixie/04-faq.md]: Maybe also reiterate that Pixie team members will not have any access to customer data. 

(We can use our support accounts only if they give us permission, but I don't know if we need to include that here).

**Review by vihangm — DISMISSED** [review]

**Review by vihangm — DISMISSED** [review]

File: content/en/03-using-pixie/03-api-quick-start.md [hunk 12]
```diff
@@ -283,3 +283,9 @@ func (s *tableMux) AcceptTable(ctx context.Context, metadata types.TableMetadata
 	return &tablePrinter{}, nil
 }
 ```
+
+## End-to-End Encryption
+
+Pixie offers end-to-end encryption for telemetry data requested by the API. For more information, see the [FAQ](/about-pixie/faq#how-does-pixie-secure-its-data).
+
+Encryption is controlled by a flag during client creation. For additional info, see the [encryption example](https://github.com/pixie-labs/pixie/blob/main/src/api/go/pxapi/examples/encryption_example/example.go#L77) project.
```
Comment (vihangm) [inline content/en/03-using-pixie/03-api-quick-start.md]: binary not project. Also should we link to a python example too?
https://github.com/pixie-labs/pixie/blob/main/src/api/python/examples/script_with_single_table.py#L35

**Review by vihangm — DISMISSED** [review]

**Review by aimichelle — APPROVED** [review]
